pkg/util/wait/wait_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wait
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGroupWait(t *testing.T) {
+	tests := []struct {
+		name          string
+		add           int
+		done          int
+		expectWaitErr bool
+	}{
+		{
+			name:          "add only",
+			add:           1,
+			expectWaitErr: true,
+		},
+		{
+			name:          "add greater than done",
+			add:           2,
+			done:          1,
+			expectWaitErr: true,
+		},
+		{
+			name:          "add equal to done",
+			add:           2,
+			done:          2,
+			expectWaitErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewGroup()
+			for i := 0; i < tt.add; i++ {
+				g.Increment()
+			}
+			for i := 0; i < tt.done; i++ {
+				g.Done()
+			}
+			err := g.WaitWithTimeout(100 * time.Millisecond)
```

Comment:
I know I always bring it up :/, but I think it's better for everything time-based if we use a virtual clock from the get go when it comes to unit tests (with the addition of a `newGroupWithClock` function). I have updated too many unit tests which would fail from time to time on Windows CI runners.

Reply:
This unit tests in this file are deterministic, but was because I didn't write some corner cases. Will rewrite it with virtual clock and add some cases.